node/node.go

```diff
@@ -335,7 +342,50 @@ func (n *Node) closeDataDir() {
 	}
 }
 
-// configureRPC is a helper method to configure all the various RPC endpoints during node
+// obtainJWTSecret loads the jwt-secret, either from the provided config,
+// or from the default location. If neither of those are present, it generates
+// a new secret and stores to the default location.
+func (n *Node) obtainJWTSecret(cliParam string) ([]byte, error) {
+	var fileName string
+	if len(cliParam) > 0 {
+		// If a plaintext secret was provided via cli flags, use that
+		jwtSecret := common.FromHex(cliParam)
+		if len(jwtSecret) == 32 && strings.HasPrefix(cliParam, "0x") {
+			log.Warn("Plaintext JWT secret provided, please consider passing via file")
+			return jwtSecret, nil
+		}
+		// path provided
+		fileName = cliParam
+	} else {
+		// no path provided, use default
+		fileName = n.ResolvePath(datadirJWTKey)
+	}
+	// try reading from file
+	log.Debug("Reading jwt-key", "path", fileName)
+	if data, err := os.ReadFile(fileName); err == nil {
```

Comment:
Does this work if there's a newline at the end? My naive guess is that ReadFile will read the newline too, which will break `common.FromHex` since it requires only hex characters (with an optional `0x` prefix).

Reply:
PS, we should support newlines at the end as many text editors dump it in there by default when you save.

Reply:
Yes exactly, this was the issue I talked about on keybase